### DISCUSSION TITLE
Not supported resolving ES modules

### DISCUSSION THREAD

**Discussion by Mori-Yang** [farm-fe/farm#1561] [Q&A]
I'm trying again to migrate from Vite to Farm（Our project is a monorepo，I'm migrating a subrepository）, I've only configured the basic inputs and outputs and when I run `npx farm build` it reports the following error:
```
Failed to load config file: Directory import '/data/home/projects/mono/apps/msg/node_modules/.farm/' is not supported resolving ES modules imported from /data/home/projects/mono/apps/msg/node_modules/.farm/farm.config.bundle-1720451557571-0d412b06867775.mjs 
```
What should I do to solve this problem? What other information would you like to know?

**Comment by fu050409** [comment]
Could you please provide the full traceback information? It will be better if you could provide a repository to reproduce this issue.

**Reply by Mori-Yang** [reply]
[root@VM-160-2 mmbizappmsg]# npx farm build
[ Farm ] Using config file at /data/home/projects/mono/apps/msg/farm.config.ts
[ Farm ] Error: Failed to load config file: Directory import '/data/home/projects/mono/apps/msg/node_modules/.farm/' is not supported resolving ES modules imported from /data/home/projects/mono/apps/msg/node_modules/.farm/farm.config.bundle-1720451557571-0d412b06867775.mjs 

Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '/data/home/projects/mono/apps/msg/node_modules/.farm/' is not supported resolving ES modules imported from /data/home/projects/mono/apps/msg/node_modules/.farm/farm.config.bundle-1720451557571-0d412b06867775.mjs
    at new NodeError (node:internal/errors:387:5)
    at finalizeResolution (node:internal/modules/esm/resolve:326:17)
    at moduleResolve (node:internal/modules/esm/resolve:907:10)
    at defaultResolve (node:internal/modules/esm/resolve:1115:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:841:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36)

**Reply by Mori-Yang** [reply]
In farm.config.bundle-1720451557571-0d412b06867775.mjs:
This line of code causes an error import __farmNodeModule from 'node:module';
I tried configuring type: "esm" in package.json and deleting node_modules to reinstall to no avail!

**Comment by wre232114** [comment]
what's in your farm.config.ts, we need more details and we'll help you finish the migration

**Reply by Mori-Yang** [reply]
This is what I'm using right now, I've commented out about some of the plugins, just these basic configurations：
```
export default defineConfig({
  compilation:{ // 编译配置
    input,
    output:{
      path:outDir,
      entryFilename:`assets/[name].${VER}[hash].js`,
      filename:`assets/[name].${VER}[hash].js`,
      assetsFilename:`assets/[name].${VER}[hash].[ext]`,
      publicPath: process.env.DEV ? LOCAL_URL : CDN_URL,
    },
    minify:false,
    css:{
      prefixer:{
        targets:['> 1%', 'ie >= 9']
      }
    },
    resolve: {
      alias: {
        "~":'',
      },
    },
  },
  // vitePlugins,
})
```

**Reply by wre232114** [reply]
could you provide a demo repo that can reproduce this issue, to help us locate the issue

**Reply by Mori-Yang** [reply]
ok, i will try to make a demo repo.
